### PR TITLE
Unify temperature sensors interface across platforms and add checks.

### PR DESCRIPTION
**Pull request by vext01**
Finishing off work I started on Frday: This essentially adds sanity checks to the sensors code, however there was some refactoring to allow the checks to be generic between platforms.

 * OpenBSD sensor code was refactored into separate "find sensors" / "read sensors" phases to match linux.
 * The sensors that are detected are now checked against the sensors that we saw prior to reboot.

OK?

### CODE REVIEW

File: krun/platform.py [hunk 1]
```diff
@@ -245,6 +253,11 @@ def bench_env_changes(self):
     def sanity_checks(self):
         pass
 
+    @abstractmethod
+    def find_temperature_sensors(self):
+        """Populates self.temp_sensors list"""
```
Comment (ltratt) [inline krun/platform.py]: Should be "populate" I think?

File: krun/platform.py [hunk 2]
```diff
@@ -376,22 +400,33 @@ def _check_apm_state(self):
...
-        return run_shell_cmd(self.TEMP_SENSORS_CMD)[0]
+        return run_shell_cmd(self.FIND_TEMP_SENSORS_CMD)[0]
+
+    def _raw_read_temperature_sensor(self, sensor):
+        # mocked in tests
+
+        # sysctl doesn't return 0 on failure.
+        # Luckily we can catch the error later when when the output
+        # looks weird.
+        return run_shell_cmd( "sysctl %s" % sensor)[0]
```
Comment (ltratt) [inline krun/platform.py]: Stray space after `(`.

File: krun/tests/test_genericplatform.py [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
         temps = {"x": 30.0, "y": 12.34, "z": 666.0}
+        mock_platform.temp_sensors = ["x", "y", "z"]
         mock_platform.starting_temperatures = temps
 
         for k, v in temps.iteritems():
             assert mock_platform.temperature_thresholds[k] == temps[k] + 5
 
+    def test_inconsistent_sensors0001(self, platform, caplog):
+        # The platform has already detected the available sensors. Now we
+        # confuse it, by moving a sensor. This shouldn't happen ofcourse, but
```
Comment (ltratt) [inline krun/tests/test_genericplatform.py]: Typo.

**Comment by ltratt** [comment]
A few minor comments.

File: krun/platform.py (hunk 1 above)
Comment (vext01) [inline krun/platform.py]: Either is fine no? As in "it populates the self.temp_sensors list".

Comment (ltratt) [inline krun/platform.py]: It's not exactly a big deal, but they have mildly different implications.

**Comment by vext01** [comment]
Fixed minor comments.

**Comment by ltratt** [comment]
OK, squash and I'll merge.

**Comment by vext01** [comment]
Plop.

**Pull request merged**